SimpleApp/ViewController.swift

```diff
@@ -2,34 +2,148 @@ import UIKit
 
 class ViewController: UIViewController {
     @IBOutlet private(set) var label: UILabel!
+    @IBOutlet weak var usernameTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    @IBOutlet weak var button: UIButton!
+    @IBOutlet weak var saveCredsSwitch: UISwitch!
+    @IBOutlet weak var savePasswordLabel: UILabel!
+
+    private enum Constants {
+        static let userNameKey = "userNameKey"
+        static let passwordKey = "passwordKey"
+        static let isCheckedKey = "isCheckedKey"
+        static let username = "user"
+        static let password = "password"
+    }
+
+    private var username: String?
+    private var password: String?
+    private var isChecked = false
+
+    @IBAction func saveCredsSwitchChanged() {
+        removeNamePasswordIfNeeded()
+    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        usernameTextField.delegate = self
+        passwordTextField.delegate = self
+        passwordTextField.isSecureTextEntry = false
+        label.text = "SimpleApp"
+        label.textAlignment = .center
+        button.setTitle("Sign In", for: .normal)
+        savePasswordLabel.text = "Save credentials"
 
-        label.text = "label"
+        usernameTextField.addTarget(
+            self,
+            action: #selector(updateUsernameProperty),
+            for: .editingChanged
+        )
+        passwordTextField.addTarget(
+            self,
+            action: #selector(updatePasswordProperty),
+            for: .editingChanged
+        )
+        restoreNamePasswordIfNeeded()
+        restoreSwitchState()
     }
-    @IBAction func buttonPressed(_ sender: UIButton) {
+
+    private func removeNamePasswordIfNeeded() {
+        let expression = !saveCredsSwitch.isOn
+        print(#function + "\(expression)")
+        if expression {
+            UserDefaults.standard.removeObject(forKey: Constants.userNameKey)
+            UserDefaults.standard.removeObject(forKey: Constants.passwordKey)
+        }
+    }
+
+    private func saveNamePasswordIfNeeded() {
+        let expression = saveCredsSwitch.isOn
+        print(#function + "\(expression)")
+        if expression {
+            username = usernameTextField.text
+            password = passwordTextField.text
```

Comment:
same for `password` , might be removed without loss any behavior

Reply:
fixed